binr/rasm2/rasm2.c

```diff
@@ -76,89 +85,107 @@ static void rasm2_list(RAsm *a, const char *arch) {
+			if (op->jump != -1LL)
+				printf ("{\"jump\": \"0x%08" PFMT64x ",", op->jump);
+			if (op->fail != -1LL)
+				printf ("{\"fail\": \"0x%08" PFMT64x ",", op->fail);
+			if (op->val != -1LL)
+				printf ("{\"value\": \"0x%08" PFMT64x ",", op->val);
+			printf ("\"stackop\": \"%s\",", stackop);
+			printf ("\"esil\": \"%s\",", r_strbuf_get (&op->esil));
+			printf ("\"stackptr\": \"0x%08" PFMT64x "\"", op->stackptr);
+			printf ("}");
+		} else {
+			printf ("offset:   0x%08" PFMT64x "\n", offset);
+			printf ("bytes:    %s\n", bytes);
+			printf ("type:     %s\n", optype);
+			if (op->jump != -1LL)
+				printf ("jump:     0x%08" PFMT64x "\n", op->jump);
+			if (op->fail != -1LL)
+				printf ("fail:     0x%08" PFMT64x "\n", op->fail);
+			//if (op->ref != -1LL)
+			//      printf ("ref:      0x%08"PFMT64x"\n", op->ref);
+			if (op->val != -1LL)
+				printf ("value:    0x%08" PFMT64x "\n", op->val);
+			printf ("stackop:  %s\n", stackop);
+			printf ("esil:     %s\n", r_strbuf_get (&op->esil));
+			printf ("stackptr: %" PFMT64d "\n", op->stackptr);
+			// produces (null) printf ("decode str: %s\n", r_anal_op_to_string (anal, op));
+			printf ("\n");
+		}
+		free (stackop);
+		free (bytes);
+	}
+	return ret;
 }
 
 static int rasm_show_help(int v) {
 	printf ("Usage: rasm2 [-ACdDehLBvw] [-a arch] [-b bits] [-o addr] [-s syntax]\n"
 		"             [-f file] [-F fil:ter] [-i skip] [-l len] 'code'|hex|-\n");
 	if (v)
-	printf (" -a [arch]    Set architecture to assemble/disassemble (see -L)\n"
-	        " -A           Show Analysis information from given hexpairs\n"
-		" -b [bits]    Set cpu register size (8, 16, 32, 64) (RASM2_BITS)\n"
-		" -c [cpu]     Select specific CPU (depends on arch)\n"
-		" -C           Output in C format\n"
-		" -d, -D       Disassemble from hexpair bytes (-D show hexpairs)\n"
-		" -e           Use big endian instead of little endian\n"
-		" -E           Display ESIL expression (same input as in -d)\n"
-		" -f [file]    Read data from file\n"
-		" -F [in:out]  Specify input and/or output filters (att2intel, x86.pseudo, ...)\n"
-		" -h           Show this help\n"
-		" -i [len]     ignore/skip N bytes of the input buffer\n"
-		" -k [kernel]  Select operating system (linux, windows, darwin, ..)\n"
-		" -l [len]     Input/Output length\n"
-		" -L           List supported asm plugins + features:\n"
-		"               a___ asm, _d__ disasm, __A_ analyzer, ___e ESIL\n"
-		" -o [offset]  Set start address for code (default 0)\n"
-		" -O [file]    Output file name (rasm2 -Bf a.asm -O a)\n"
-		" -s [syntax]  Select syntax (intel, att)\n"
-		" -B           Binary input/output (-l is mandatory for binary input)\n"
-		" -v           Show version information\n"
-		" -w           What's this instruction for? describe opcode\n"
-		" If '-l' value is greater than output length, output is padded with nops\n"
-		" If the last argument is '-' reads from stdin\n");
+		printf (" -a [arch]    Set architecture to assemble/disassemble (see -L)\n"
```

Comment:
indent.sh is not magic, but if you used this script, then whitelist it :P

> On 30 Mar 2016, at 16:29, Maijin <notifications@github.com> wrote:
> 
> In binr/rasm2/rasm2.c <https://github.com/radare/radare2/pull/4458#discussion_r57897930>:
> 
> > -		" -F [in:out]  Specify input and/or output filters (att2intel, x86.pseudo, ...)\n"
> > -		" -h           Show this help\n"
> > -		" -i [len]     ignore/skip N bytes of the input buffer\n"
> > -		" -k [kernel]  Select operating system (linux, windows, darwin, ..)\n"
> > -		" -l [len]     Input/Output length\n"
> > -		" -L           List supported asm plugins + features:\n"
> > -		"               a___ asm, _d__ disasm, __A_ analyzer, ___e ESIL\n"
> > -		" -o [offset]  Set start address for code (default 0)\n"
> > -		" -O [file]    Output file name (rasm2 -Bf a.asm -O a)\n"
> > -		" -s [syntax]  Select syntax (intel, att)\n"
> > -		" -B           Binary input/output (-l is mandatory for binary input)\n"
> > -		" -v           Show version information\n"
> > -		" -w           What's this instruction for? describe opcode\n"
> > -		" If '-l' value is greater than output length, output is padded with nops\n"
> > -		" If the last argument is '-' reads from stdin\n");
> > +		printf (" -a [arch]    Set architecture to assemble/disassemble (see -L)\n"
> Would be cool if that would be done by ident.sh :)
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/radare/radare2/pull/4458/files/f870dbd0d7fd48513a1578ef218d6a8594e85c2d#r57897930>